main.js

```diff
@@ -48,3 +50,140 @@ navbarBtn.addEventListener('click', () => {
+    linkLive: 'https://waelelsafty07.github.io/Portfolio/',
+    linkSource: 'https://github.com/waelelsafty07/Portfolio',
+  },
+];
+
+const repeatList = (list, text) => {
+  const fragment = document.createDocumentFragment();
+
+  list.forEach((item, index) => {
+    const li = document.createElement('li');
+    if (text === 'info') {
+      if (index === 0) li.className = 'db-color none-style';
+      else li.className = 'bg-color';
+    }
+
+    const liText = document.createTextNode(item);
+    li.appendChild(liText);
+    fragment.appendChild(li);
+  });
+  return fragment;
+};
+
+const createProject = (project) => {
+  const projectDiv = document.createElement('div');
+  projectDiv.className = 'project';
+  const cardDiv = document.createElement('div');
+  cardDiv.className = 'card';
+  const cardImageDiv = document.createElement('div');
+  cardImageDiv.className = 'card-image';
+  const img = document.createElement('img');
+  img.src = project.image;
+  cardImageDiv.appendChild(img);
+  cardDiv.appendChild(cardImageDiv);
+  const cardBodyDiv = document.createElement('div');
+  cardBodyDiv.className = 'card-body';
+  const BodyTitleDiv = document.createElement('div');
+  BodyTitleDiv.className = 'card-title';
+  const h2 = document.createElement('h2');
+  const h2Text = document.createTextNode(project.name);
+  h2.appendChild(h2Text);
+  BodyTitleDiv.appendChild(h2);
+  cardBodyDiv.appendChild(BodyTitleDiv);
+  const ClientInfoDiv = document.createElement('div');
+  ClientInfoDiv.className = 'card-client-info';
+  const ul = document.createElement('ul');
+  const itemlist = ['CANOPY', 'Back End Dev', '2015'];
+  const lists = repeatList(itemlist, 'info');
+  ul.appendChild(lists);
+  ClientInfoDiv.appendChild(ul);
+  cardBodyDiv.appendChild(ClientInfoDiv);
+
+  const BodyPragrphDiv = document.createElement('div');
+  BodyPragrphDiv.className = 'card-pragrph';
+  const p = document.createElement('p');
+  const pText = document.createTextNode(project.description);
+  p.appendChild(pText);
+  BodyPragrphDiv.appendChild(p);
+  cardBodyDiv.appendChild(BodyPragrphDiv);
+
+  const CardTechDiv = document.createElement('div');
+  CardTechDiv.className = 'card-tech';
+  const ulTech = document.createElement('ul');
+  const listTech = project.technologies;
+  const listTechs = repeatList(listTech);
+  ulTech.appendChild(listTechs);
+  CardTechDiv.appendChild(ulTech);
+  cardBodyDiv.appendChild(CardTechDiv);
+
+  const CardBtnDiv = document.createElement('div');
+  CardBtnDiv.className = 'card-btn';
+  const CardBtn = document.createElement('button');
+  const buttonText = document.createTextNode('See project');
+  CardBtn.appendChild(buttonText);
+  CardBtnDiv.appendChild(CardBtn);
+  cardBodyDiv.appendChild(CardBtnDiv);
+
+  cardDiv.appendChild(cardBodyDiv);
+  projectDiv.appendChild(cardDiv);
+  work.appendChild(projectDiv);
+};
+
+Projects.forEach((project) => {
+  createProject(project);
+});
```

Comment:
[OPTIONAL] Well done overall 💯 but before calling `createProject` for each project in the `Projects` array, you could add a check to ensure that all the required fields (such as `name`, `description`, and `image`) are present in each project object. If any of the required fields are missing, you could throw an error or log a warning to the console. ⚠️